rules/Dbal40/Rector/StmtsAwareInterface/ExecuteQueryParamsToBindValueRector.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\Dbal40\Rector\StmtsAwareInterface;
+
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\BinaryOp\Plus;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Foreach_;
+use PhpParser\NodeFinder;
+use Rector\Contract\PhpParser\Node\StmtsAwareInterface;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class ExecuteQueryParamsToBindValueRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Change executeQuery() with parameters to bindValue() with explicit values', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Doctrine\DBAL\Statement;
+
+class SomeClass
+{
+    public function run(Statement $statement, array $params): void
+    {
+        $result = $statement->executeQuery($params)
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Doctrine\DBAL\Statement;
+
+class SomeClass
+{
+    public function run(Statement $statement, array $params): void
+    {
+        foreach ($params as $key=> $value) {
+            $statement->bindValue($key + 1, $value);
+        }
+
+        $result = $statement->executeQuery();
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StmtsAwareInterface::class];
+    }
+
+    /**
+     * @param StmtsAwareInterface $node
+     */
+    public function refactor(Node $node): ?StmtsAwareInterface
+    {
+        $nodeFinder = new NodeFinder();
+
+        $hasChanged = false;
+
+        foreach ((array) $node->stmts as $key => $stmt) {
+            if (! $stmt instanceof Expression) {
+                continue;
+            }
+
+            $executeQueryMethodCall = $nodeFinder->findFirst($stmt, function (Node $node): bool {
+                if (! $node instanceof MethodCall) {
+                    return false;
+                }
+
+                if (! $this->isName($node->name, 'executeQuery')) {
```

Comment:
this needs check `executeQuery` came from `Doctrine\DBAL\Statement` object to avoid different object changed. I will create PR for it.

Reply:
https://github.com/rectorphp/rector-doctrine/pull/413